### PR TITLE
Reimplemented fall particles

### PR DESCRIPTION
**Pull request by tigerw**
Regression when 1.9.x support introduced in #3135, originally implemented in #461.

### CODE REVIEW

File: src/Entities/Pawn.cpp [hunk 1]
```diff
@@ -428,9 +428,14 @@ void cPawn::HandleFalling(void)
...
 			// Fall particles
-			// TODO: Re-enable this when effects in 1.9 aren't broken (right now this uses the wrong effect ID in 1.9 and the right one in 1.8)
-			// int ParticleSize = static_cast<int>((std::min(15, Damage) - 1.f) * ((50.f - 20.f) / (15.f - 1.f)) + 20.f);
-			// GetWorld()->BroadcastSoundParticleEffect(EffectID::PARTICLE_FALL_PARTICLES, POSX_TOINT, POSY_TOINT - 1, POSZ_TOINT, ParticleSize);
+			GetWorld()->BroadcastParticleEffect(
+				"blockdust",
+				GetPosition(),
+				{ 0, 0, 0 },
+				(Damage - 1.f) * ((0.3f - 0.1f) / (15.f - 1.f)) + 0.1f,  // Map damage (1 - 15) to particle speed (0.1 - 0.3)
+				static_cast<int>((Damage - 1.f) * ((50.f - 20.f) / (15.f - 1.f)) + 20.f),  // Map damage (1 - 15) to particle quantity (20 - 50)
```
Comment (NiLSPACE) [inline src/Entities/Pawn.cpp]: Where did we get the math from?

File: src/Entities/Pawn.cpp [hunk 2]
```diff
@@ -428,9 +428,14 @@ void cPawn::HandleFalling(void)
 			TakeDamage(dtFalling, nullptr, Damage, Damage, 0);
 
 			// Fall particles
-			// TODO: Re-enable this when effects in 1.9 aren't broken (right now this uses the wrong effect ID in 1.9 and the right one in 1.8)
-			// int ParticleSize = static_cast<int>((std::min(15, Damage) - 1.f) * ((50.f - 20.f) / (15.f - 1.f)) + 20.f);
-			// GetWorld()->BroadcastSoundParticleEffect(EffectID::PARTICLE_FALL_PARTICLES, POSX_TOINT, POSY_TOINT - 1, POSZ_TOINT, ParticleSize);
+			GetWorld()->BroadcastParticleEffect(
```
Comment (peterbell10) [inline src/Entities/Pawn.cpp]: This function already exists by the way:

https://github.com/cuberite/cuberite/blob/31a11a6df4922b590a50a5ff3d3c00d42a45599d/src/Broadcaster.h#L11-L13

You have to get to it through a `cBroadcaster`
```cpp
GetWorld()->GetBroadcaster().BroadcastParticleEffect([...]);
```

File: src/Entities/Pawn.cpp (hunk 1 above)
Comment (tigerw) [reply src/Entities/Pawn.cpp]: Technically https://github.com/cuberite/cuberite/commit/de2cce9b79ab60427daa86dcd1da99db7d7ea161#diff-ab7bd0f99b0b93eb1655e0775b322933 but I just wrote code to map one range of numbers to another. Both ranges were guesses but looks about right.

Also, m a t h e m a t i c
(use maths, it's superior ;))

**Comment by tigerw** [comment]
The build failed for networking reasons. I wonder why.

**Comment by peterbell10** [comment]
Looks like #4124, only happens on apple for some reason.

**Pull request merged**